models/sla_policy/reply_time/int_zendesk__reply_time_combined.sql

```diff
@@ -119,7 +120,8 @@ with reply_time_calendar_hours_sla as (
       and ((
         agent_reply_at >= sla_schedule_start_at and agent_reply_at <= sla_schedule_end_at) -- ticket is replied to between a schedule window
         or (agent_reply_at < sla_schedule_start_at and sum_lapsed_business_minutes_new = 0 and sla_breach_at = first_sla_breach_at) -- ticket is replied to before a schedule window and no business minutes have been spent on it
-        or (agent_reply_at is null and {{ dbt.current_timestamp() }} >= sla_schedule_start_at and {{ dbt.current_timestamp() }} < next_schedule_start) -- ticket is not replied to and therefore active. But only bring through the active SLA record that is most recent (after the last SLA schedule starts but before the next)  
+        or (agent_reply_at is null and {{ dbt.current_timestamp() }} >= sla_schedule_start_at and ({{ dbt.current_timestamp() }} < next_schedule_start or next_schedule_start is null)) -- ticket is not replied to and therefore active. But only bring through the active SLA record that is most recent (after the last SLA schedule starts but before the next, or if there does not exist a next SLA schedule start time)  
+        or (agent_reply_at >= sla_schedule_end_at and agent_reply_at <= next_schedule_start ) -- ticket is replied to outside sla schedule hours
```

Comment:
Should this be exclusive as opposed to inclusive? For example, what if a ticket is replied to at the exact same time the sla_schedule is ending? Could this possible bring in duplicates then?

I am wondering if this needs to be the following instead.
```suggestion
        or (agent_reply_at > sla_schedule_end_at and agent_reply_at < next_schedule_start ) -- ticket is replied to outside sla schedule hours
```

Reply:
Thanks for catching. Yes since the 1st condition (within a schedule window) is inclusive, then this ought to be exclusive since this is the contrary (outside the window) to avoid potential duplicates for reply times right on top of the schedule window. Updated!